### PR TITLE
Fix FP S3800 (`function-return-type`): Exception for sanitation functions

### LINKED ISSUE #4527: Fix FP S3800 (`function-return-type`): Exceptions for React components and sanitation functions
Add exception for

- React components, ignore when returning `null` or `false`
- sanitation functions: when 2 return types: string and boolean

### PR DESCRIPTION
**Pull request by ericmorand-sonarsource**
Fixes #4527

### CODE REVIEW

File: its/ruling/src/test/expected/jsts/vuetify/javascript-S3800.json [hunk 1]
```diff
@@ -1,37 +1,5 @@
 {
 "vuetify:packages/docs/src/examples/v-date-picker/event-events.vue": [
 44
-],
```
Comment (ericmorand-sonarsource) [inline its/ruling/src/test/expected/jsts/vuetify/javascript-S3800.json:4]: For information, I checked all the removed lines and can confirm that they are not triggering the rule anymore - they all belong to a sanitation function.

File: packages/jsts/src/rules/S3800/unit.test.ts [hunk 2]
```diff
@@ -468,13 +491,6 @@ ruleTesterTs.run(`Functions should always return the same type [ts]`, rule, {
         },
       ],
     },
-    {
```
Comment (ericmorand-sonarsource) [inline packages/jsts/src/rules/S3800/unit.test.ts:471]: This test was moved from the invalid list to the valid one.

File: packages/jsts/src/rules/S3800/unit.test.ts [hunk 3]
```diff
@@ -533,6 +549,14 @@ ruleTesterTs.run(`Functions should always return the same type [ts]`, rule, {
         },
       ],
     },
+    {
+      code: `
```
Comment (ericmorand-sonarsource) [inline packages/jsts/src/rules/S3800/unit.test.ts:553]: I added this test to guarantee that only functions with `true` and `string` return types are valid.

File: its/ruling/src/test/expected/jsts/vuetify/javascript-S3800.json (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply its/ruling/src/test/expected/jsts/vuetify/javascript-S3800.json:4]: 👍

File: packages/jsts/src/rules/S3800/unit.test.ts (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply packages/jsts/src/rules/S3800/unit.test.ts:471]: 👍

File: packages/jsts/src/rules/S3800/unit.test.ts (hunk 3 above)
Comment (yassin-kammoun-sonarsource) [reply packages/jsts/src/rules/S3800/unit.test.ts:553]: 👍

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**